test/correctness/atomics.cpp

```diff
@@ -533,74 +506,64 @@ void test_hist_compute_at(const Backend &backend) {
-            } else {
-                hist.update()
-                    .atomic()
-                    .split(r, ro, ri, 8)
-                    .parallel(ro)
-                    .vectorize(ri);
-            }
-        } break;
-        case Backend::OpenCL: {
-            RVar ro, ri;
-            if (is_float_16) {
-                // Associativity prover doesn't support float16.
-                // Set override_associativity_test to true to remove the check.
-                hist.update().atomic(true /*override_associativity_test*/).split(r, ro, ri, 32)
-                    .gpu_blocks(ro, DeviceAPI::OpenCL)
-                    .gpu_threads(ri, DeviceAPI::OpenCL);
-            } else {
-                hist.update().atomic().split(r, ro, ri, 32)
-                    .gpu_blocks(ro, DeviceAPI::OpenCL)
-                    .gpu_threads(ri, DeviceAPI::OpenCL);
-            }
-        } break;
-        case Backend::CUDA: {
-            RVar ro, ri;
-            hist.update().atomic().split(r, ro, ri, 32)
-                .gpu_blocks(ro, DeviceAPI::CUDA)
-                .gpu_threads(ri, DeviceAPI::CUDA);
-        } break;
-        case Backend::CUDAVectorize: {
-            RVar ro, ri;
-            RVar rio, rii;
-            hist.update().atomic()
-                .split(r, ro, ri, 32).split(ri, rio, rii, 4)
-                .gpu_blocks(ro, DeviceAPI::CUDA)
-                .gpu_threads(rio, DeviceAPI::CUDA)
-                .vectorize(rii);
-        } break;
-        default: {
-            _halide_user_assert(false) << "Unsupported backend.\n";
-        } break;
+    switch (backend) {
+    case Backend::CPU: {
+        if (is_float_16) {
+            // Associativity prover doesn't support float16.
+            // Set override_associativity_test to true to remove the check.
+            hist.update().atomic(true /*override_associativity_test*/).parallel(r);
+        } else {
+            hist.update().atomic().parallel(r);
+        }
+    } break;
+    case Backend::CPUVectorize: {
+        RVar ro, ri;
+        if (is_float_16) {
+            // Associativity prover doesn't support float16.
+            // Set override_associativity_test to true to remove the check.
+            hist.update()
+                .atomic(true /*override_associativity_test*/)
+                .split(r, ro, ri, 8)
+                .parallel(ro)
+                .vectorize(ri);
+        } else {
+            hist.update()
+                .atomic()
+                .split(r, ro, ri, 8)
+                .parallel(ro)
+                .vectorize(ri);
+        }
+    } break;
+    case Backend::OpenCL: {
+        RVar ro, ri;
+        if (is_float_16) {
+            // Associativity prover doesn't support float16.
+            // Set override_associativity_test to true to remove the check.
+            hist.update().atomic(true /*override_associativity_test*/).split(r, ro, ri, 32).gpu_blocks(ro, DeviceAPI::OpenCL).gpu_threads(ri, DeviceAPI::OpenCL);
```

Comment:
Also here. Looks like the problem was that the original did .update().atomic() a lot, so it might be easiest to roll back, fix that, then re-clang-format